### PR TITLE
Implement CRUD operations on sequences

### LINKED ISSUE #292: Scenario viewer - Possibility to add/delete/rename sequences in scenarios
## New feature

Sequences can be added and deleted at runtime. The goal would be to allow users to create sequences through the Scenario viewer.

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves #292

Update to the scenario viewer control

### CODE REVIEW

File: frontend/taipy/src/ScenarioViewer.tsx [hunk 1]
```diff
@@ -108,38 +116,52 @@ const tagsAutocompleteSx = {
     maxWidth: "none",
 };
 
+type SequenceFull = [string, string, string[], boolean, boolean];
+// enum SeFProps {
```
Comment (FabienLelaquais) [inline frontend/taipy/src/ScenarioViewer.tsx:120]: Do you really want to keep these?

File: taipy/gui_core/_context.py [hunk 2]
```diff
@@ -411,26 +411,52 @@ def edit_entity(self, state: State, id: str, payload: t.Dict[str, str]):
...
+                    if data.get("del", False):
+                        if scenario := t.cast(Scenario, core_get(entity.owner_id)):
+                            scenario.remove_sequence(entity._get_name())
+                    else:
+                        name = data.get(_GuiCoreContext.__PROP_ENTITY_NAME)
+                        tasks = data.get("task_ids")
+                        if entity._get_name() != name:
+                            if scenario := t.cast(Scenario, core_get(entity.owner_id)):
+                                with scenario:
+                                    scenario.add_sequence(name, tasks)
```
Comment (FabienLelaquais) [inline taipy/gui_core/_context.py]: What if the new name is the name of an existing sequence?

**Review by FabienLelaquais — DISMISSED** [review]
Looks great. Just have to put that in gear.
Thx

File: frontend/taipy/src/ScenarioViewer.tsx (hunk 1 above)
Comment (FredLL-Avaiga) [reply frontend/taipy/src/ScenarioViewer.tsx:120]: yes, I think it's important to describe somewhere the format

File: taipy/gui_core/_context.py (hunk 2 above)
Comment (FredLL-Avaiga) [reply taipy/gui_core/_context.py]: Should core handle that ?

Comment (FredLL-Avaiga) [reply taipy/gui_core/_context.py]: Will handle it

File: taipy/gui_core/_context.py [hunk 3]
```diff
@@ -446,11 +446,8 @@ def edit_entity(self, state: State, id: str, payload: t.Dict[str, str]):
                         tasks = data.get("task_ids")
                         if entity._get_name() != name:
                             if scenario := t.cast(Scenario, core_get(entity.owner_id)):
-                                with scenario:
-                                    scenario.add_sequence(name, tasks)
-                                    scenario.remove_sequence(entity._get_name())
-                                    entity = scenario.sequences[name]
-                        else:
+                                scenario.rename_sequence(entity._get_name(), name)
```
Comment (FabienLelaquais) [inline taipy/gui_core/_context.py]: I'm worried not to see any exception catch here...

**Review by FabienLelaquais — DISMISSED** [review]
One worrying line...

File: taipy/gui_core/_context.py (hunk 3 above)
Comment (FredLL-Avaiga) [reply taipy/gui_core/_context.py]: https://vscode.dev/github/Avaiga/taipy/blob/feature/%23292-CRUD-sequence/taipy/gui_core/_context.py#L425

**Review by FabienLelaquais — APPROVED** [review]
🤔

**Pull request merged**